workspaces/ballerina/data-mapper/src/components/Diagram/Label/ExpressionLabelWidget.tsx

```diff
@@ -234,10 +238,15 @@ export function ExpressionLabelWidget(props: ExpressionLabelWidgetProps) {
         // TODO: Add impl
     }
 
-    if (mappingType !== MappingType.Default) {
+    codeActions.push({
+        title: "Map with custom function",
+        onClick: onClickMapWithCustomFn
+    });
+
+    if (mappingType !== MappingType.Default) {   
```

Comment:
`mappingType !== MappingType.Default` will provide this option for `union` type scenarios as well isnt'it?

Shall we explicitly provide this option only for record/array-to-record/array ?